test/native/testRunner.hpp

```diff
@@ -13,11 +13,23 @@
 #include <string>
 #include <vector>
 
-struct TestCase;
+struct TestCase {
+    std::string name;
+    std::function<void()> test_function;
+    bool only; // only run this test when true, ignore all others.
+    std::string filename;
+    int line_no;
+    int assertion_count = 0;
+    bool has_failed_assertions = false;
+    bool skipped = false;
+
+    TestCase(const std::string& name, std::function<void()> test_function, bool only, const std::string& filename, int line_no)
+        : name(name), test_function(test_function), only(only), filename(filename), line_no(line_no) {}
+};
 
 class TestRunner {
   private:
-    std::map<std::string, TestCase> _test_cases;
+    std::unordered_map<std::string, TestCase> _test_cases;
```

Comment:
In this case, running tests in a consistent order is useful.

Reply:
6a640e0dc29a52ced53bec0c9a2346ad19c59d80